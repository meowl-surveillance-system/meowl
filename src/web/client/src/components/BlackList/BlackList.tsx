import React, { Component } from "react";
import { Container, TextField, Button } from "@material-ui/core";

interface Props {
    name: String,
    isLoggedIn: boolean
}
interface State {}

/** 
 * A component for rendering one notification
*/
export default class BlackList extends Component<Props, State> {
    constructor(props: Props) {
        super(props);
    }

    /**
     * Renders a single notification
     */
    render() {
        const info = this.props;
        return (
            <Container>
                <form>
                    <TextField label="name"/>
                    <Button> Submit </Button>
                </form>
            </Container>
        );
    }
}
