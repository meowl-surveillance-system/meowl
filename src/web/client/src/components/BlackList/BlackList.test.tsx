import React from "react";
import { render } from "@testing-library/react";
import BlackList from "./BlackList";

describe('Notification component', () => {
  it("renders Notification component", () => {
    const { container } = render(
      <BlackList isLoggedIn={true} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
}) 