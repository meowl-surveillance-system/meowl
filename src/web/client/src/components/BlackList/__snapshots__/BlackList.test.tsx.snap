// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component renders Notification component 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg"
>
  <form
    novalidate=""
  >
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Insert the name of the individual you would like to blacklist (all lowercase)
    </p>
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
      >
        name
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input"
          name="name"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
         Submit 
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
